apps/manager_app/device_authentication.c

```diff
@@ -0,0 +1,367 @@
+                                               manager_result_t *result,
+                                               bool *result_valid);
+
+/**
+ * @brief This function is the request handler for the SIGN_RANDOM_NUM state of
+ * the device authentication flow
+ *
+ * @param query Reference to query of type manager_query_t as received from the
+ * host
+ * @param result Reference to the result of type manager_result_t, which will be
+ * populated by the API. This API will only fill the auth_device field of the
+ * manager_result_t.
+ * @param result_valid If the result returned from the API is valid.
+ * @return device_auth_state_e The next state of the flow
+ */
+static device_auth_state_e sign_random_handler(const manager_query_t *query,
+                                               manager_result_t *result,
+                                               bool *result_valid);
+
+/**
+ * @brief This function is the request handler for the RESULT state of
+ * the device authentication flow
+ *
+ * @param query Reference to query of type manager_query_t as received from the
+ * host
+ * @param result Reference to the result of type manager_result_t, which will be
+ * populated by the API. This API will only fill the auth_device field of the
+ * manager_result_t.
+ * @param result_valid If the result returned from the API is valid.
+ * @return device_auth_state_e The next state of the flow
+ */
+static device_auth_state_e result_handler(const manager_query_t *query,
+                                          manager_result_t *result,
+                                          bool *result_valid);
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static pb_size_t get_request_type(
+    const manager_auth_device_request_t *request) {
+  return request->which_request;
+}
+
+static manager_auth_device_response_t send_flow_complete(void) {
+  manager_auth_device_response_t response =
+      MANAGER_AUTH_DEVICE_RESPONSE_INIT_ZERO;
+  response.which_response = MANAGER_AUTH_DEVICE_RESPONSE_FLOW_COMPLETE_TAG;
+  response.flow_complete.dummy_field = '\0';
+  return response;
+}
+
+static device_auth_state_e sign_serial_handler(const manager_query_t *query,
+                                               manager_result_t *result,
+                                               bool *result_valid) {
+  pb_size_t request_type = get_request_type(&query->auth_device);
+  device_auth_state_e next_state = SIGN_SERIAL_NUM;
+
+  switch (request_type) {
+    case MANAGER_AUTH_DEVICE_REQUEST_INITIATE_TAG: {
+      result->auth_device = sign_serial_number();
+      *result_valid = true;
+      next_state = SIGN_RANDOM_NUM;
+      break;
+    }
+    case MANAGER_AUTH_DEVICE_REQUEST_CHALLENGE_TAG:
+    case MANAGER_AUTH_DEVICE_REQUEST_RESULT_TAG:
+    default: {
+      usb_clear_event();
+      break;
+    }
```

Comment:
Set *result_valid = false here;
```suggestion
    default: {
      *result_valid = false;
      usb_clear_event();
      break;
    }
```